src/Controller/Controller.php

```diff
@@ -823,7 +823,7 @@ public function isAction(string $action): bool
      * @return \Cake\Http\Response|null|void
      * @link https://book.cakephp.org/4/en/controllers.html#request-life-cycle-callbacks
      */
-    public function beforeFilter(EventInterface $event)
+    public function beforeFilter(EventInterface $event): Response|null|null
```

Comment:
This was skipped because we can declare a type with `|void`.

Reply:
I don't think we should add return types to event callback methods. It would be quite annoying to unnecessary add `return null` in most cases.